### PR TITLE
Switch to go modules

### PR DESCRIPTION
**Pull request by ibuildthecloud**
~This PR has the terrible downside that we use git submodules
for kubernetes.  I'm not sure a better way to make this work short
of copying kubernetes into this repo as a subdirectory.~

### CODE REVIEW

**Comment by MarkusTeufelberger** [comment]
Do you really need to ship the full vendor folder with this repository?

**Comment by warmchang** [comment]
Deprecate the [Trash - Go ./vendor manager](https://github.com/rancher/trash) tool?

**Comment by ibuildthecloud** [comment]
@warmchang Yes we want to move away from `trash`

@MarkusTeufelberger Switching to go modules will be done in two phased.  The first is to get go modules managing the vendor directory.  In this phase CI and build scripts still use a classic GOPATH setup and need vendor.  The next step is to drop vendor directory, rely on go proxy, and move all build scripts to go modules.  Phase 2 is TBD.

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+	github.com/kubernetes-sigs/cri-tools => github.com/rancher/cri-tools v1.15.0-k3s.2
+	github.com/matryer/moq => github.com/rancher/moq v0.0.0-20190404221404-ee5226d43009
+	github.com/opencontainers/runtime-spec => github.com/opencontainers/runtime-spec v0.0.0-20180911193056-5684b8af48c1
+	github.com/rancher/dynamiclistener => github.com/erikwilson/rancher-dynamiclistener v0.0.0-20190717164634-c08b499d1719
+	github.com/rancher/kine => github.com/ibuildthecloud/kine v0.1.0
+	golang.org/x/sys => github.com/golang/sys v0.0.0-20190204203706-41f3e6584952
+	k8s.io/api => ./kubernetes/staging/src/k8s.io/api
+	k8s.io/apiextensions-apiserver => ./kubernetes/staging/src/k8s.io/apiextensions-apiserver
+	k8s.io/apimachinery => ./kubernetes/staging/src/k8s.io/apimachinery
+	k8s.io/apiserver => ./kubernetes/staging/src/k8s.io/apiserver
```
Comment (yamt) [inline go.mod]: looking at https://github.com/golang/go/wiki/modules#publishing-a-release ,
i guess ```k8s.io/apiserver => github.com/rancher/k3s/staging/src/k8s.io/apiserver v1.15.3-k3s.1``` works if you add tags like ```staging/src/k8s.io/apiserver/v1.15.3-k3s.1```

Comment (ibuildthecloud) [reply go.mod]: Thanks! Let me try this.

Comment (ibuildthecloud) [reply go.mod]: @yamt Thanks so much, this worked. I can drop the submodule now.

**Comment by ibuildthecloud** [comment]
@erikwilson This is GTG

**Comment by ibuildthecloud** [comment]
Ugh, we are going to move k8s to github.com/rancher/kubernetes

**Pull request merged**